matlab/test/arrow/type/tListType.m

```diff
@@ -0,0 +1,124 @@
+% TLISTTYPE Tests for arrow.type.ListType
+
+% Licensed to the Apache Software Foundation (ASF) under one or more
+% contributor license agreements.  See the NOTICE file distributed with
+% this work for additional information regarding copyright ownership.
+% The ASF licenses this file to you under the Apache License, Version
+% 2.0 (the "License"); you may not use this file except in compliance
+% with the License.  You may obtain a copy of the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS,
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+% implied.  See the License for the specific language governing
+% permissions and limitations under the License.
+
+classdef tListType < matlab.unittest.TestCase
+
+    properties (Constant)
+        BasicList = arrow.list(arrow.int8())
+        NestedList = arrow.list(arrow.list(arrow.list(arrow.uint64())))
+        ConstructionFcn = @arrow.list
+        TypeID = arrow.type.ID.List
+        ClassName = "arrow.type.ListType"
+    end
+
+    methods (Test)
+
+        function ConstructionFcnBasic(testCase)
+            % Verify construction function returns an instance of the
+            % expected arrow.type.Type subclass.
+            type = testCase.BasicList;
+            testCase.verifyInstanceOf(type, testCase.ClassName);
+        end
+
+        function ConstructionFcnTooFewInputsError(testCase)
+            % Verify construction function errors if given zero input arguments.
+            fcn = @() testCase.ConstructionFcn();
+            testCase.verifyError(fcn, "MATLAB:minrhs");
+        end
+
+        function ConstructionFcnInvalidInputTypeError(testCase)
+            % Verify construction function errors if any one of the input
+            % arguments is not an arrow.type.Type object.
+            fcn = @() testCase.ConstructionFcn("abc");
+            testCase.verifyError(fcn, "MATLAB:validation:UnableToConvert");
+        end
+
+        function ConstructionFcnEmptyTypeError(testCase)
+            % Verify construction function errors if given an empty
+            % arrow.type.Type array as one of its inputs.
```

Comment:
as one of its inputs -> as the input argument.

Reply:
Thanks for catching this! This was a copy/paste mistake.